hibernate-core/src/test/java/org/hibernate/orm/test/lob/JarFileEntryBlobTest.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright Red Hat Inc. and Hibernate Authors
+ */
+package org.hibernate.orm.test.lob;
+
+import org.hibernate.bugs.TestEntity;
+import org.hibernate.engine.jdbc.env.internal.NonContextualLobCreator;
+import org.hibernate.testing.orm.junit.DomainModel;
+import org.hibernate.testing.orm.junit.JiraKey;
+import org.hibernate.testing.orm.junit.SessionFactory;
+import org.hibernate.testing.orm.junit.SessionFactoryScope;
+import org.junit.jupiter.api.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.sql.Blob;
+import java.sql.SQLException;
+import java.util.jar.JarEntry;
+import java.util.jar.JarFile;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+@DomainModel(annotatedClasses = TestEntity.class)
+@SessionFactory
+@JiraKey( "HHH-19464" )
+class JarFileEntryBlobTest {
+
+	@Test
+	void hibernate_blob_streaming(SessionFactoryScope scope) throws Exception {
+		final var zipFilePath = getClass().getClassLoader().getResource( "org/hibernate/orm/test/lob/JarFileEntryBlobTest.zip" );
```

Comment:
Not great (and potentially dangerous) to have a `zip` file just for testing purposes. Could we reproduce the error with a simple byte-array and `org.hibernate.engine.jdbc.proxy.BlobProxy`?

Reply:
Any stream that will always read complete buffer (with exception of last read) will not expose this problem. `java.util.zip.InflaterInputStream` can return less bytes before reading next bytes to be deflated.

Possible way to avoid zip file in resources is to create it temporary on th fly.